test/algorithms/buffer/buffer_variable_width.cpp

```diff
@@ -0,0 +1,184 @@
+// Boost.Geometry
+
+// Copyright (c) 2024 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include <geometry_test_common.hpp>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/optional.hpp>
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+
+namespace bg = boost::geometry;
+
+#if defined(TEST_WITH_SVG)
+#include "test_buffer_svg.hpp"
+#endif
+
+// A point with extra info, such that
+// - it can influence the buffer with dynamically (input)
+// - it can receive the side at which the buffer was (output)
+struct specific_point
+{
+    double x{0.0};
+    double y{0.0};
+    double distance_left{0.0};
+    double distance_right{0.0};
+    boost::optional<bg::strategy::buffer::buffer_side_selector> side{};
+};
+
+BOOST_GEOMETRY_REGISTER_POINT_2D(specific_point, double, cs::cartesian, x, y)
+
+struct specific_buffer_side_strategy
+{
+    bool equidistant() const
+    {
+        return true;
+    }
+
+    template <typename Point, typename OutputRange, typename DistanceStrategy>
+    bg::strategy::buffer::result_code apply(Point const& input_p1, Point const& input_p2,
+                                            bg::strategy::buffer::buffer_side_selector side,
+                                            DistanceStrategy const& distance,
+                                            OutputRange &output_range) const
+    {
+        const auto result =
```

Comment:
Maybe `auto const` which seems that is more common as a style in BG? 
Also why not `auto const&`?

Reply:
Yes it is. But on my office it's not, so I sometimes don't switch at the right moment...
Thanks.
BTW, in this case it should not be `&` because it's an assignment from a function result. (Though, technically, it's in line with C++ specs)

Reply:
:white